tests/pytests/unit/modules/test_gpg.py

```diff
@@ -0,0 +1,797 @@
+        with patch.dict(gpg.__salt__, {"config.option": mock_opt}):
+            with patch.object(gpg, "_list_keys", return_value=_list_result):
+                ret = gpg.get_key("xxxxxxxxxxxxxxxx")
+                assert ret == _expected_result
+
+
+def test_delete_key_without_passphrase(gpghome):
+    """
+    Test gpg.delete_key without passphrase
+    """
+
+    _user_mock = {
+        "shell": "/bin/bash",
+        "workphone": "",
+        "uid": 0,
+        "passwd": "x",
+        "roomnumber": "",
+        "gid": 0,
+        "groups": ["root"],
+        "home": str(gpghome.path),
+        "fullname": "root",
+        "homephone": "",
+        "name": "root",
+    }
+
+    _list_result = [
+        {
+            "dummy": "",
+            "keyid": "xxxxxxxxxxxxxxxx",
+            "expires": "2011188692",
+            "sigs": [],
+            "subkeys": [
+                ["xxxxxxxxxxxxxxxx", "e", "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"]
+            ],
+            "length": "4096",
+            "ownertrust": "-",
+            "sig": "",
+            "algo": "1",
+            "fingerprint": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+            "date": "1506612692",
+            "trust": "-",
+            "type": "pub",
+            "uids": ["GPG Person <person@example.com>"],
+        }
+    ]
+
+    _expected_result = {
+        "res": True,
+        "message": "Secret key for xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx deleted\nPublic key for xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx deleted",
+    }
+
+    mock_opt = MagicMock(return_value="root")
+    with patch.dict(gpg.__salt__, {"user.info": MagicMock(return_value=_user_mock)}):
+        with patch.dict(gpg.__salt__, {"config.option": mock_opt}):
+            with patch.object(gpg, "_list_keys", return_value=_list_result):
+                with patch(
+                    "salt.modules.gpg.gnupg.GPG.delete_keys",
+                    MagicMock(return_value="ok"),
+                ) as gnupg_delete_keys:
+                    ret = gpg.delete_key(
+                        "xxxxxxxxxxxxxxxx", delete_secret=True, use_passphrase=False
+                    )
+                    assert ret == _expected_result
+                    gnupg_delete_keys.assert_called_with(
+                        "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+                        False,
+                        expect_passphrase=False,
+                    )
+
+
+def test_delete_key_with_passphrase_without_gpg_passphrase_in_pillar(gpghome):
+    """
+    Test gpg.delete_key with passphrase but without gpg_passphrase pillar
+    """
+
+    _user_mock = {
+        "shell": "/bin/bash",
+        "workphone": "",
+        "uid": 0,
+        "passwd": "x",
+        "roomnumber": "",
+        "gid": 0,
+        "groups": ["root"],
+        "home": str(gpghome.path),
+        "fullname": "root",
+        "homephone": "",
+        "name": "root",
+    }
```

Comment:
It's late now, since this has been merged already, but the fact that this `_user_mock` is identical for several (alll?) of these tests means that could be extracted to fixtures. Getting patching out of tests and into fixtures often times clarifies exactly what is being tested.

Reply:
:+1: